botocore/signers.py

```diff
@@ -188,31 +195,49 @@ def generate_presigned_url(self, request_dict, expires_in=3600,
         """
         if region_name is None:
             region_name = self._region_name
-        query_prefix = '-query'
+        query_suffix = '-query'
         signature_version = self._signature_version
-        if not signature_version.endswith(query_prefix):
-            signature_version += query_prefix
+        if signature_version is not botocore.UNSIGNED and not \
+                signature_version.endswith(query_suffix):
+            signature_version += query_suffix
 
-        kwargs = {'signing_name': self._signing_name,
-                  'region_name': region_name,
-                  'signature_version': signature_version,
-                  'expires': expires_in}
+        handler, response = self._event_emitter.emit_until_response(
```

Comment:
I think we should consolidate the event emission into one helper function to make sure that if we update the event at all we account for all places.